### ISSUE TITLE
Text jumping when using Markdown

### ISSUE THREAD

**Issue by sandmule** [data-maki/flowtoken#2]
When using the animated markdown component, the text briefly shifts off-screen during generation before correcting itself. This jump is jarring, though the text component functions as expected.

Any help is much appreciated

<img width="808" alt="Screenshot 2024-10-21 at 16 42 45" src="https://github.com/user-attachments/assets/e160f8a0-aa7b-42b8-a786-9a44ce245766">

**Comment by Ephibbs** [comment]
Sorry you encountered an issue! Can you tell me what browser you are using as well as how you are using the react component? I'll try to replicate it on my end.

**Comment by Ephibbs** [comment]
@sandmule Just following up. Are you setting any global styles that are affecting the container or spans? It may be messing with the layout on first render. Please let me know if you found a fix and I can make the library more robust.

**Comment by sandmule** [comment]
Hi sorry for the delay. It was a animation issue caused by an upwrapped string. I added this to a fork to fix it. I haven't tested in isolation so could be a css issue in my setup.
```
    li: ({ node, ...props }: any) => {
      const childrenArray = React.Children.toArray(props.children);
      const isStringContent = (child: any) => typeof child === 'string';

      // Check if the content is a valid inline element, such as <strong> or <em>
      const isValidInlineNode = (child: any) => typeof child === 'object' && ['strong', 'em', 'code', 'a', 'span'].includes(child?.props.node.tagName);

      // Only wrap in <p> if the content is string or a valid inline node
      const needsWrapping = childrenArray.every((child: any) => isStringContent(child) || isValidInlineNode(child));
      // Only wrap the content in <p> if it's a string

      return (
        <li {...props} className="custom-li" style={animationStyle}>
          {needsWrapping ? <p>{animateText(props.children)}</p> : animateText(props.children)}
        </li>
      );
```

**Comment by Ephibbs** [comment]
I made some changes and released a new version 1.0.18. Please try that one. I discovered a small bug that was causing unnecessary rerenders which I believe is related to this issue.
